### ISSUE TITLE
Implement menu link and contents in mobile view

### ISSUE THREAD

**Issue by gsidebo** [mitodl/mitxonline#23]
Depends on #21 

Designs:
- Hamburger menu: https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/455498563
- Menu contents, logged in: https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/455498564
- Menu contents, logged out: https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/455488638

**Acceptance criteria:**
- At mobile screen width, only a hamburger menu is shown on the right side of the site header, and it matches the designs 
- Clicking the hamburger menu changes the icon to an "X", an overlay appears over the body of the site, and the correct options are shown depending on whether the user is logged in or logged out
- Click the "X" icon hides the overlay menu and changes the icon back to a hamburger menu icon

**Not needed:**
- "Change password" link – Right now, this form is included in the Settings page. If we implement this as a separate page, that can be done in a separate issue
